qiskit_machine_learning/algorithms/classifiers/pegasos_qsvc.py

```diff
@@ -0,0 +1,285 @@
+        """
+        # check whether the data have the right format
+        if np.ndim(X) != 2:
+            raise ValueError("X has to be a 2D array")
+        if np.ndim(y) != 1:
+            raise ValueError("y has to be a 1D array")
+        if len(np.unique(y)) != 2:
+            raise ValueError("Only binary classification is supported")
+        if X.shape[0] != y.shape[0]:
+            raise ValueError("'X' and 'y' have to contain the same number of samples")
+        if self._precomputed and X.shape[0] != X.shape[1]:
+            raise ValueError(
+                "For a precomputed kernel, X should be in shape (n_samples, n_samples)"
+            )
+        if sample_weight is not None:
+            raise NotImplementedError(
+                "Parameter 'sample_weight' is not supported. All samples have to be weighed equally"
+            )
+        # reset the fit state
+        self._fit_status = False
+
+        # the algorithm works with labels in {+1, -1}
+        self._label_pos = np.unique(y)[0]
+        self._label_neg = np.unique(y)[1]
+        self._label_map = {self._label_pos: +1, self._label_neg: -1}
+
+        # the training data are later needed for prediction
+        self._x_train = X
+        self._y_train = y
+        self._n_samples = X.shape[0]
+
+        # empty dictionary to represent sparse array
+        self._alphas = {}
+
+        t_0 = datetime.now()
+        # training loop
+        for step in range(1, self._num_steps + 1):
+            # for every step, a random index (determining a random datum) is fixed
+            i = np.random.randint(0, len(y))
+
+            value = self._compute_weighted_kernel_sum(i, X, training=True)
+
+            if (self._label_map[y[i]] * self.C / step) * value < 1:
+                # only way for a component of alpha to become non zero
+                self._alphas[i] = self._alphas.get(i, 0) + 1
+
+        self._fit_status = True
+
+        logger.debug("fit completed after %s", str(datetime.now() - t_0)[:-7])
+
+        return self
+
+    # pylint: disable=invalid-name
+    def predict(self, X: np.ndarray) -> np.ndarray:
+        """
+        Perform classification on samples in X.
+
+        Args:
+            X: Features. For a callable kernel shape (m_samples, n_features), for a precomputed
+               kernel shape (m_samples, n_samples), where m denotes the set to be predicted and n the
+               size of the training set. In that case, the kernel values in X have to be calculated
+               with respect to the elements of the set to be predicted and the training set.
+
+        Returns:
+            y_pred: Shape (n_samples), the predicted class labels for samples in X.
+
+        Raises:
+            QiskitMachineLearningError:
+                - predict is called before the model has been fit.
+            ValueError:
+                - Pre-computed kernel matrix has the wrong shape and/or dimension.
+        """
+        if not self._fit_status:
```

Comment:
I see sklearn in its SVCs has a `fit_status` attribute - does this integrate in someway with that - or should it. To me here it looks standalone but I am not an expert on sklearn

Reply:
They have `fit_status_`. Unless we want to make it as a public API, I don't think we should re-use their property.

Reply:
I guess my comment was more that this sub-class seems to have its own status and SVC as a class had its. Since this is a derived class from SVC I would expect all its methods etc to be applicable - they may be overridden of course to be more specialized. So if I call fit should I not be able to look at fit_status_ and get the status as I do if I use the SVC directly - or is that aspect of SVC broken here. Should this then be extending SVC - I see there is a classifermixin but maybe thats not enough.

Reply:
The problem is that there's no a clean interface in `sklearn` we could derive from. `ClassifierMixin` does not define all methods like `fit/predict/score`, it has only `score`. Other methods are defined in `BaseLibSVM`. We only use method definitions here, that's why we extend `SVC`. This may not be the best way and I'm happy to change this. 

Perhaps, we should design our own base interface for classifiers/regressors that resembles `sklearn` and derive all classes from it. In this case we would have full compatibility in the interface, but things `isinstance` would not work.

Reply:
It seems that in extending SVC you want to be able to use this, as a specialized version of an SVC, in contexts where SVC can be used, which makes a great deal of sense. I.e. has the same public API/interface and in fact some of the methods etc only exist in the parent. Maybe there are other aspects that will mean that a simple swap does not work in practice depending on the code in question though. For this example I happened to notice, around fit_status, if you checked that in the code you had been using with the SVC then things are not going to work as expected with this algorithm. In the case of fit_status it might just be simple to use the same instance var - the only place I see it being set is after the call to libsvm.fit So maybe linking to this is not so bad in practice though of course things might change with it in the future. Now this is the aspect I noticed and commented on - I saw other attributes etc but I did not look further. Perhaps mostly in practice these are aspects that are infrequently used and most people would be able to use this algorithm, how they were using an SVC in sklearn, without noticing such. Its not like things can't be changed - we can certainly see how this goes and whether we get any feedback in these regards.

Reply:
OK, replaced `_fit_status` with `fit_status_`, I doubt this is a good choice.